util/argo/argo.go

```diff
@@ -20,20 +28,53 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
-
-	argoappv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
-	"github.com/argoproj/argo-cd/v2/pkg/client/clientset/versioned/typed/application/v1alpha1"
-	applicationsv1 "github.com/argoproj/argo-cd/v2/pkg/client/listers/application/v1alpha1"
-	"github.com/argoproj/argo-cd/v2/reposerver/apiclient"
-	"github.com/argoproj/argo-cd/v2/util/db"
-	"github.com/argoproj/argo-cd/v2/util/io"
-	"github.com/argoproj/argo-cd/v2/util/settings"
 )
 
 const (
 	errDestinationMissing = "Destination server missing from app spec"
 )
 
+// FormatSyncMsg enrich the K8s message with user-relevant information
+func FormatSyncMsg(res *common.ResourceSyncResult, apiVersFn func() ([]kube.APIResourceInfo, error)) error {
+
+	switch res.Message {
+	case "the server could not find the requested resource":
+
+		resource, err := getAPIResourceInfo(res.ResourceKey.Kind, apiVersFn)
+		if err != nil {
+			return err
+		}
+		res.Message = fmt.Sprintf("The server could not find requested resource %s. Make sure the CRD is installed on the destination cluster, "+
+			"and that the requested CRD version is available. Currently, the installed API version for the corresponding Kind: %s is %s",
+			res.ResourceKey.Name, resource.GroupKind.Kind, resource.GroupVersionResource.Version)
```

Comment:
I think we can be more helpful in cases where the CRD isn't installed.

```suggestion
		resource, err := getAPIResourceInfo(res.ResourceKey.Kind, apiVersFn)
		if err != nil {
			return err
		}
		if resource == nil {
			res.Message = fmt.Sprintf("The Kubernetes API could not find %s/%s for requested resource %s/%s. Make sure the %q CRD is installed on the destination cluster.", res.ResourceKey.Group, res.ResourceKey.Kind, res.ResourceKey.Namespace, res.ResourceKey.Name, res.ResourceKey.Kind)
		} else {
			res.Message = fmt.Sprintf("The Kubernetes API could not find version %q of %s/%s for requested resource %s/%s. Version %q of %s/%s is installed on the destination cluster.", res.Version, res.ResourceKey.Group, res.ResourceKey.Kind, res.ResourceKey.Namespace, res.ResourceKey.Name, resource.GroupVersionResource.Version, resource.GroupKind.Group, resource.GroupKind.Kind)"
		}
```